discovery_engine/lib/src/ffi/types/history.dart

```diff
@@ -52,3 +54,30 @@ extension HistoricDocumentFfi on HistoricDocument {
     );
   }
 }
+
+final _listFfiAdapter = ListFfiAdapter(
+  alloc: ffi.alloc_uninitialized_historic_document_slice,
+  next: ffi.next_historic_document,
+  writeNative: (doc, place) => doc.writeNative(place),
+  readNative: HistoricDocumentFfi.readNative,
+  getVecLen: ffi.get_historic_document_vec_len,
+  getVecBuffer: ffi.get_historic_document_vec_buffer,
+  writeNativeVec: ffi.init_historic_document_vec_at,
+);
+
+extension HistoricDocumentSliceFfi on List<HistoricDocument> {
```

Comment:
isn't this the `HistoricDocumentVecFfi`? in contrast to our other dart `*SliceFfi`s this only contains vec functionality.
```suggestion
extension HistoricDocumentVecFfi on List<HistoricDocument> {
```
maybe we should either include slice and vec functionality for all `*SliceFfi`s or we could split the extensions into a `*SliceFfi` and a `*VecFfi`.

Reply:
Yes, I renamed it.